demo/src/screens/MainScreen.js

```diff
@@ -174,17 +220,80 @@ class MainScreen extends Component {
         hideUnderline
         floatingPlaceholder={false}
         text70
-        trailingAccessory={
+        leadingAccessory={
           filterText ? (
-            <Button link iconSource={Assets.icons.demo.close} $iconDefault onPress={this.clearSearch}/>
+            <Button link marginR-5 iconSource={Assets.icons.demo.close} $iconDefault onPress={this.clearSearch}/>
           ) : (
-            <Icon tintColor={Colors.$iconDefault} source={Assets.icons.demo.search}/>
+            <Icon marginR-10 tintColor={Colors.$iconDefault} source={Assets.icons.demo.search}/>
           )
         }
       />
     );
   };
 
+  onPress = ({customValue: index}) => {
+    const {chipsLabels} = this.state;
+    this.onPressFlag = true;
+    this.onUserScroll = false;
+    this.setState({
+      selectedSection: index,
+      faderStart: index != 0,
+      faderEnd: index !== chipsLabels.length - 1
+    });
+  };
+
+  onCheckViewableItems = ({viewableItems}) => {
+    const {chipsLabels, selectedSection} = this.state;
+    if (this.onPressFlag === false && this.onUserScroll === true) {
```

Comment:
comparing a boolean value to `false` is similar to adding a not (!) before it. 
For example instead of `this.onPressFlag === false` you can do `!this.onPressFlag`

As for comparing a boolean value to `true` - this is redundant. 

a boolean value is an expression on its own